src/include/cephfs/libcephfs.h

```diff
@@ -164,6 +164,19 @@ void ceph_userperm_destroy(UserPerm *perm);
  */
 struct UserPerm *ceph_mount_perms(struct ceph_mount_info *cmount);
 
+/**
+ * Set cmount's default permissions
+ *
+ * @param cmount the mount info handle
+ * @param perm permissions to set to default for mount
+ *
+ * Every cmount has a default set of credentials. This sets copies the given
+ * permissions to the ones in the cmount. Must be done after ceph_init
+ *
+ * Returns 0 on success, and -EISCONN if the cmount is already mounted.
+ */
+int ceph_mount_perms_set(struct ceph_mount_info *cmount, UserPerm *perm);
```

Comment:
The UserPerm* should be const'ed all over if we're not taking ownership of it, right?

Reply:
Probably so. I'm thinking I may just drop the top two patches altogether though since we're not removing the conf options now.